actions/graybox/promote-worker.js

```diff
@@ -67,89 +67,114 @@ async function main(params) {
-                        const content = await response.text();
-                        let docx;
-                        const { sp } = await getConfig();
-
-                        if (content.includes(experienceName) || content.includes(gbStyleExpression) || content.includes(gbDomainSuffix)) {
-                            docx = await updateDocument(content, experienceName, helixAdminApiKey);
-                            if (docx) {
-                                // Save file Destination full path with file name and extension
-                                const destinationFilePath = `${stat.path.substring(0, stat.path.lastIndexOf('/') + 1).replace('/'.concat(experienceName), '')}${stat.fileName}`;
-
-                                const saveStatus = await saveFile(docx, destinationFilePath, sp);
-                                if (saveStatus && saveStatus.success === true) {
-                                    promoteStatuses.push(destinationFilePath);
-                                } else {
-                                    failedPromoteStatuses.push(destinationFilePath);
-                                }
-                            } else {
-                                logger.error(`Error generating docx file for ${stat.path}`);
-                            }
-                        } else {
-                            const copySourceFilePath = `${stat.path.substring(0, stat.path.lastIndexOf('/') + 1)}${stat.fileName}`; // Copy Source full path with file name and extension
-                            const copyDestinationFolder = `${stat.path.substring(0, stat.path.lastIndexOf('/')).replace('/'.concat(experienceName), '')}`; // Copy Destination folder path, no file name
-                            const promoteCopyFileStatus = await promoteCopy(copySourceFilePath, copyDestinationFolder, stat.fileName, sp);
+        // Update Promote Status
+        const sPromoteStatuses = promotes.length > 0 ? `Successful Promotes: \n${promotes.join('\n')}` : '';
+        const sFailedPromoteStatuses = failedPromotes.length > 0 ? `Failed Promotes: \n${failedPromotes.join('\n')}` : '';
+        const promoteExcelValues = [['Promote', toUTCStr(curreDateTime), sPromoteStatuses]];
+        const failedPromoteExcelValues = [['Promote', toUTCStr(curreDateTime), sFailedPromoteStatuses]];
 
-                            if (promoteCopyFileStatus) {
-                                promoteStatuses.push(`${copyDestinationFolder}/${stat.fileName}`);
-                            } else {
-                                failedPromoteStatuses.push(`${copyDestinationFolder}/${stat.fileName}`);
-                            }
-                        }
-                    }
-                });
-            }
-        });
+        if (promotes.length > 0) {
+            await updateExcelTable(projectExcelPath, 'PROMOTE_STATUS', promoteExcelValues, IS_GRAYBOX);
```

Comment:
Is this writing the successful paths to excel? That might be problematic if there are 100s of files.. It will be good to limit it to just failed paths with a status of completion. 
Also, given that  the code was ported to graybox specific repo, I don't think the `IS_GRAYBOX` boolean is needed

Reply:
Sure Sunil. Thanks!

Reply:
Yes it is writing the list of promotes, will update to hold only failures.

Reply:
Updated to include the trigger actions, and failures, and removed the successful promotes and is_graybox option